src/Utils.cpp

```diff
@@ -180,6 +178,32 @@ unsigned long CUtils::GetLongIP(const CString& sIP) {
     return ret;
 }
 
+CString CUtils::GetHostName() {
+    const char *pEnv;
+
+    pEnv = getenv("HOSTNAME");
+    if (pEnv && pEnv[0])
+        return pEnv;
+
+#if defined(HAVE_GETHOSTNAME) && defined(HAVE_SYSCONF)
+    long HostNameMax = sysconf(_SC_HOST_NAME_MAX);
```

Comment:
How is this better than `HOST_NAME_MAX` macro?

Reply:
Probably it is not. I'll fix that with a buffer on stack

Reply:
It is not always clear what to expect when dealing with these constants. Same applies, for instance, to `PATH_MAX`.
Anyway, I used `_POSIX_HOST_NAME_MAX` because it seems to build both under FreeBSD and Linux. It is not the case for `HOST_NAME_MAX`.

Reply:
We have CI for linux, macos, cygwin and freebsd, though freebsd is currently only running for pushes on master branch, not for PRs